plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -512,4 +503,19 @@ private static TableIdentifier toIdentifier(SchemaTableName schemaTableName)
     {
         return TableIdentifier.of(schemaTableName.getSchemaName(), schemaTableName.getTableName());
     }
+
+    private List<Namespace> relevantNamespaces(ConnectorSession session, Optional<String> namespace)
+    {
+        List<Namespace> namespaces = ImmutableList.of();
+        if (namespace.isEmpty()) {
+            namespaces = listNamespaces(session).stream()
+                    .map(Namespace::of)
+                    .collect(toImmutableList());
+        }
+        else if (namespaceExists(session, namespace.get())) {
```

Comment:
the `else` will become redundant, remove it

Reply:
Addressed in https://github.com/trinodb/trino/pull/19980